augraphy/augmentations/voronoi.py

```diff
@@ -0,0 +1,180 @@
+
+    @staticmethod
+    @jit(nopython=True, cache=True)
+    def generate_voronoi(width, height, num_cells, nsize, pixel_data, perlin_noise_2d):
+        """
+        Generates Voronoi Tessellation Image
+        """
+        img_array = np.zeros((width, height), dtype=np.uint8)
+        for y in range(width):
+            for x in range(height):
+                dmin = math.hypot(height, width)
+                for i in range(num_cells):
+                    d = math.hypot(
+                        (pixel_data[0][i] - x + perlin_noise_2d[0][x][y]),
+                        (pixel_data[1][i] - y + perlin_noise_2d[1][x][y]),
+                    )
+                    if d < dmin:
+                        dmin = d
+                        j = i
+                    nsize[j] += 1
+                img_array[y][x] = pixel_data[2][j]
+        return img_array
+
+    def apply_augmentation(self):
+        obj_noise = Noise()
+        perlin_x = np.zeros((self.height, self.width))
+        perlin_y = np.zeros((self.height, self.width))
+        if self.perlin:
+            perlin_x = np.array(
+                [
+                    [obj_noise.noise2D(x / 100, y / 100) * self.mult for y in range(self.height)]
+                    for x in range(self.width)
+                ],
+            )
+            perlin_y = np.array(
+                [
+                    [
+                        obj_noise.noise2D((x + self.seed) / 100, (y + self.seed) / 100) * self.mult
+                        for y in range(self.height)
+                    ]
+                    for x in range(self.width)
+                ],
+            )
+        nx = [random.randrange(self.width) for _ in range(self.num_cells)]  # x-coordinates of random points
+        ny = [random.randrange(self.height) for _ in range(self.num_cells)]  # y-coordinates of random points
+        ng = [
+            random.randrange(self.background_value[0], self.background_value[1]) for _ in range(self.num_cells)
+        ]  # go through the number of cells and assign color
+        nsize = np.zeros(self.num_cells, dtype=np.int32)
+        img_array = self.generate_voronoi(
+            self.width,
+            self.height,
+            self.num_cells,
+            nsize,
+            (nx, ny, ng),
+            (perlin_x, perlin_y),
+        )
+        image = Image.fromarray(img_array)
+        image.save("images/Voronoi_example.png", "PNG", dpi=(300, 300))
+        mesh = cv2.imread("images/Voronoi_example.png")
+        os.remove("images/Voronoi_example.png")
+        return mesh
+
+    # Applies the Augmentation to input data.
+    def __call__(self, image, layer=None, force=False):
+        if force or self.should_run():
+            result = image.copy()
+            h, w, _ = result.shape
```

Comment:
This line should be replaced by `h, w = result.shape[:2]`, and line 179 below can be just :

    result = result[self.ws : h + self.ws, self.ws : w + self.ws]

We might facing error with the additional `:` if the input is in grayscale.